cdc/capture/http_util.go

```diff
@@ -0,0 +1,243 @@
+		if err != cerror.ErrStartTsBeforeGC {
+			return nil, cerror.ErrPDEtcdAPIError.Wrap(err)
+		}
+		return nil, err
+	}
+
+	// verify target-ts
+	if changefeedConfig.TargetTS > 0 && changefeedConfig.TargetTS <= changefeedConfig.StartTS {
+		return nil, cerror.ErrTargetTsBeforeStartTs.GenWithStackByArgs(changefeedConfig.TargetTS, changefeedConfig.StartTS)
+	}
+
+	// init replicaConfig
+	replicaConfig := config.GetDefaultReplicaConfig()
+	replicaConfig.ForceReplicate = changefeedConfig.ForceReplicate
+	if changefeedConfig.MounterWorkerNum != 0 {
+		replicaConfig.Mounter.WorkerNum = changefeedConfig.MounterWorkerNum
+	}
+	if changefeedConfig.SinkConfig != nil {
+		replicaConfig.Sink = changefeedConfig.SinkConfig
+	}
+	if len(changefeedConfig.IgnoreTxnStartTs) != 0 {
+		replicaConfig.Filter.IgnoreTxnStartTs = changefeedConfig.IgnoreTxnStartTs
+	}
+	if len(changefeedConfig.FilterRules) != 0 {
+		replicaConfig.Filter.Rules = changefeedConfig.FilterRules
+	}
+
+	// set sortEngine and EnableOldValue
+	_, captureInfos, err := capture.etcdClient.GetCaptures(ctx)
+	if err != nil {
+		return nil, cerror.ErrPDEtcdAPIError.Wrap(err)
+	}
+	cdcClusterVer, err := version.GetTiCDCClusterVersion(captureInfos)
+	if err != nil {
+		return nil, err
+	}
+	sortEngine := model.SortUnified
+	if !cdcClusterVer.ShouldEnableOldValueByDefault() {
+		replicaConfig.EnableOldValue = false
+		log.Warn("The TiCDC cluster is built from unknown branch or less than 5.0.0-rc, the old-value are disabled by default.")
+		if !cdcClusterVer.ShouldEnableUnifiedSorterByDefault() {
+			sortEngine = model.SortInMemory
+		}
+	}
+
+	// init ChangefeedInfo
+	info := &model.ChangeFeedInfo{
+		SinkURI:           changefeedConfig.SinkURI,
+		Opts:              make(map[string]string),
+		CreateTime:        time.Now(),
+		StartTs:           changefeedConfig.StartTS,
+		TargetTs:          changefeedConfig.TargetTS,
+		Config:            replicaConfig,
+		Engine:            sortEngine,
+		State:             model.StateNormal,
+		SyncPointEnabled:  false,
+		SyncPointInterval: 10 * time.Minute,
+		CreatorVersion:    version.ReleaseVersion,
+	}
+
+	if !replicaConfig.ForceReplicate && !changefeedConfig.IgnoreIneligibleTable {
+		ineligibleTables, _, err := verifyTables(capture, replicaConfig, changefeedConfig.StartTS)
+		if err != nil {
+			return nil, err
+		}
+		if len(ineligibleTables) != 0 {
+			return nil, cerror.ErrTableIneligible.GenWithStackByArgs(ineligibleTables)
+		}
+	}
+
+	err = verifySink(ctx, info.SinkURI, info.Config, info.Opts)
+	if err != nil {
+		return nil, err
+	}
+
+	return info, nil
+}
+
+func verifyTables(capture *Capture, replicaConfig *config.ReplicaConfig, startTs uint64) (ineligibleTables, eligibleTables []model.TableName, err error) {
+	meta, err := kv.GetSnapshotMeta(capture.kvStorage, startTs)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	filter, err := filter.NewFilter(replicaConfig)
```

Comment:
better not use the same name with "github.com/pingcap/ticdc/pkg/filter"

Reply:
done